### ISSUE TITLE
[BUG] URL encoded username/password values in inet:url nodes aren't decoded when parsed to their respective properties

### ISSUE THREAD

**Issue by captainGeech42** [vertexproject/synapse#2568]
**Describe the bug**
Currently, when you create a `inet:url` node, the values that get parsed out of the `:base` value for the `:user` and `:passwd` props are taken as literal values. However, for cases where a restricted character (e.g., `@`) is present in one of those values, it must be URL encoded to properly be sent to the server.

From my perspective, it makes sense to URL decode the `:user` and `:passwd` properties so that the resulting `inet:user` and `inet:passwd` nodes can be pivoted on more widely for cases where they may be observed without URL encoding.

**To Reproduce**
```
> [inet:url="smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587"]
inet:url=smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587
	.created = 2022/02/26 19:28:14.569
	:base = smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587
	:fqdn = smtp.gmail.com
	:params = 
	:passwd = p%40ssw0rd1
	:path = 
	:port = 587
	:proto = smtp
	:user = myemail%40domain.com
```
The username and password properties still have the `@` character URL encoded, preventing further pivoting.

**Expected behavior**
```
> [inet:url="smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587"]
inet:url=smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587
	.created = 2022/02/26 19:28:14.569
	:base = smtp://myemail%40domain.com:p%40ssw0rd1@smtp.gmail.com:587
	:fqdn = smtp.gmail.com
	:params = 
	:passwd = p@ssw0rd1
	:path = 
	:port = 587
	:proto = smtp
	:user = myemail@domain.com
```
The `@` in the username and password properties has been decoded properly, but is still URL encoded in the `:base` value since that's how it was literally observed.

**Environment (please complete the following information):**
 - Synapse 2.82.1
 - Optic 2.37.0

**Comment by aaronst** [comment]
I think #3146 would do it @captainGeech42, but I'm not super familiar with the code base.

**Comment by aaronst** [comment]
The test in `tests/test_model_inet.py` makes me think that this behavior is desired though.

```python
        # rfc3986 compliant Userinfo with @ properly encoded
        url = f'calrissian://visi%40vertex.link:surround%40@{host_port}:44343'
        expected = (f'calrissian://visi%40vertex.link:surround%40@{repr_host_port}:44343', {'subs': {
            'proto': 'calrissian', 'path': '',
            'user': 'visi%40vertex.link', 'passwd': 'surround%40',
            'base': f'calrissian://visi%40vertex.link:surround%40@{repr_host_port}:44343',
            'port': 44343,
            'params': '',
            htype: norm_host,
        }})
        self.eq(t.norm(url), expected)
```

**Comment by invisig0th** [comment]
I'm not sure that `visi%40vertex.link` is actually a valid encoding for a URL. Thanks for bringing this up, we'll def have a look.

**Comment by invisig0th** [comment]
Whups, misread ( that's what i get for reading tickets before enough coffee. That's def a valid URL encoding, but you're def correct that the username/password should be URL decoded by the URL parser 👍

**Comment by vEpiphyte** [comment]
Resolved by #3169  and will be included in v2.137.0.
